### PR TITLE
Scaffolding Arcade Migration

### PR DESCRIPTION
**Pull request by deepchoudhery**
Migrating the build from korebuild to arcade.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -33,17 +59,18 @@
...
-    <MicrosoftExtensionsDependencyInjectionPackageVersion>3.0.0</MicrosoftExtensionsDependencyInjectionPackageVersion>
+    <MicrosoftExtensionsDependencyInjectionPackageVersion>3.1.0-preview1.19506.1</MicrosoftExtensionsDependencyInjectionPackageVersion>
 
     <!-- Microsoft.Extensions.FileProviders.Physical -->
-    <MicrosoftExtensionsFileProvidersPhysicalPackageVersion>3.0.0</MicrosoftExtensionsFileProvidersPhysicalPackageVersion>
+    <MicrosoftExtensionsFileProvidersPhysicalPackageVersion>3.1.0-preview1.19506.1</MicrosoftExtensionsFileProvidersPhysicalPackageVersion>
 
     <!-- TFM -->
-    <MicrosoftNETCoreApp30PackageVersion>3.0.0-preview1-26907-05</MicrosoftNETCoreApp30PackageVersion>
+    <MicrosoftNETCoreApp30PackageVersion>3.0.0</MicrosoftNETCoreApp30PackageVersion>
```
Comment (vijayrkn) [inline eng/Versions.props]: This should not be used.

File: eng/Versions.props [hunk 2]
```diff
@@ -33,17 +59,18 @@
...
 
     <NewtonsoftJsonPackageVersion>11.0.2</NewtonsoftJsonPackageVersion>
     <NuGetFrameworksPackageVersion>4.7.0</NuGetFrameworksPackageVersion>
     <VisualStudio_NewtonsoftJsonPackageVersion>9.0.1</VisualStudio_NewtonsoftJsonPackageVersion>
```
Comment (vijayrkn) [inline eng/Versions.props]: We should check what version of newtonsoft vs ships.

File: src/VS.Web.CG.Design/VS.Web.CG.Design.csproj [hunk 3]
```diff
@@ -9,13 +9,19 @@
     <PackageTags>aspnetcore;codegenerator;scaffolding;visualstudioweb</PackageTags>
     <PackageId>Microsoft.VisualStudio.Web.CodeGeneration.Design</PackageId>
     <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp3.0' ">win7-x64</RuntimeIdentifier>
+    <X86BinDirectory>..\..\artifacts\bin\VS.Web.CG.Design-x86\</X86BinDirectory>
```
Comment (vijayrkn) [inline src/VS.Web.CG.Design/VS.Web.CG.Design.csproj]: Change to well known artifactsbindir msbuild property.

File: src/VS.Web.CG.Design/VS.Web.CG.Design.csproj [hunk 4]
```diff
@@ -9,13 +9,19 @@
...
+    <AnyCpuBinDirectory>..\..\artifacts\bin\VS.Web.CG.Design-anycpu\</AnyCpuBinDirectory>
+
     <X86ProjectDirectory>..\..\tools\VS.Web.CG.Design-x86\</X86ProjectDirectory>
     <ARMProjectDirectory>..\..\tools\VS.Web.CG.Design-arm\</ARMProjectDirectory>
     <ARM64ProjectDirectory>..\..\tools\VS.Web.CG.Design-arm64\</ARM64ProjectDirectory>
     <AnyCpuProjectDirectory>..\..\tools\VS.Web.CG.Design-anycpu\</AnyCpuProjectDirectory>
     <IncludeBuildOutput>false</IncludeBuildOutput>
     <NuspecFile>$(MSBuildProjectName).nuspec</NuspecFile>
     <IncludeSymbols>true</IncludeSymbols>
+    <packageIconFullPath>..\..\eng\packageIcon.png</packageIconFullPath>
```
Comment (vijayrkn) [inline src/VS.Web.CG.Design/VS.Web.CG.Design.csproj]: This should go to package.props

File: src/VS.Web.CG.Design/VS.Web.CG.Design.csproj [hunk 5]
```diff
@@ -27,61 +33,51 @@
...
+      <PackageReference Update="Microsoft.NETCore.App" PrivateAssets="All" />
   </ItemGroup>
 
   <Target Name="PopulateNuspec" BeforeTargets="GenerateNuspec" DependsOnTargets="Build">
 
     <PropertyGroup>
       <!-- Make sure we create a symbols.nupkg. -->
       <IncludeSymbols>true</IncludeSymbols>
-
       <NuspecProperties>
```
Comment (vijayrkn) [inline src/VS.Web.CG.Design/VS.Web.CG.Design.csproj]: We should remove the nuspec & use msbuild properties & Ispackable.

File: src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiControllerWithContext.cshtml [hunk 6]
```diff
@@ -56,8 +56,6 @@ namespace @Model.ControllerNamespace
         }
 
         // PUT: @routePrefix/5
-        // To protect from overposting attacks, please enable the specific properties you want to bind to, for
```
Comment (vijayrkn) [inline src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiControllerWithContext.cshtml]: Any src files except csproj should be removed from this PR.

File: src/dotnet-aspnet-codegenerator/dotnet-aspnet-codegenerator.csproj [hunk 7]
```diff
@@ -1,5 +1,5 @@
-<Project Sdk="Internal.AspNetCore.Sdk">
-
+﻿<Project Sdk="Microsoft.Net.Sdk">
+    <Import Project="$(RepositoryEngineeringDir)\Package.props" />
```
Comment (vijayrkn) [inline src/dotnet-aspnet-codegenerator/dotnet-aspnet-codegenerator.csproj]: All projects that generated packages should import this and common properties should move to this.

File: test/Directory.Build.props [hunk 8]
```diff
@@ -4,14 +4,13 @@
...
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkPackageVersion)" />
     <PackageReference Include="Moq" Version="$(MoqPackageVersion)" />
-    <PackageReference Include="xunit" Version="$(XunitPackageVersion)" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="$(XunitRunnerVisualStudioPackageVersion)" />
     <PackageReference Include="xunit.extensibility.execution" Version="$(XunitExtensibilityExecutionPackageVersion)" />
     <PackageReference Include="xunit.skippablefact" Version="$(XunitSkippableFactPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="3.0.0" />
```
Comment (vijayrkn) [inline test/Directory.Build.props]: Version should be from Version.props

File: test/E2E_Test/MSBuildProjectStrings.cs [hunk 9]
```diff
@@ -0,0 +1,985 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (vijayrkn) [inline test/E2E_Test/MSBuildProjectStrings.cs]: There should only be one file and all projects should share.

File: test/TestApps/ClassLibrary1/ClassLibrary1.csproj [hunk 10]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <IsPackable>false</IsPackable>
```
Comment (vijayrkn) [inline test/TestApps/ClassLibrary1/ClassLibrary1.csproj]: You can create a TestPackage.props and all test projects can import them.

File: Aracde Migration.txt [hunk 11]
```diff
@@ -0,0 +1,6 @@
+//TODO
```
[the remark this answers is not in the document]
Comment (deepchoudhery) [reply Aracde Migration.txt]: deleted for now. will add to wiki when done.

File: Directory.Build.props [hunk 12]
```diff
@@ -1,20 +1,23 @@
...
+  <Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
 
-  <Import Project="version.props" />
-  <Import Project="build\dependencies.props" />
-  <Import Project="build\sources.props" />
+  <PropertyGroup Condition="'$(CopyrightMicrosoft)' != ''">
+    <Copyright>$(CopyrightMicrosoft)</Copyright>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <IsPackable>true</IsPackable>
+    <DebugType>full</DebugType>
```
[the remark this answers is not in the document]
Comment (deepchoudhery) [reply Directory.Build.props]: changed to pdbonly. embedded does not produce the pdb we need for nupkg.

**Pull request merged**